larray_editor/editor.py

```diff
@@ -275,6 +278,26 @@ def get_value(self):
         # already been destroyed, due to the Qt.WA_DeleteOnClose attribute
         return self.data
 
+    def save_widgets_state_and_geometry(self):
+        settings = QSettings()
+        settings.setValue('geometry/{}'.format(self.name), self.saveGeometry())
+        settings.setValue('state/{}'.format(self.name), self.saveState())
+        for widget_name, widget in self.widget_state_settings.items():
+            settings.setValue('state/{}'.format(widget_name), widget.saveState())
```

Comment:
I don't like mixing "user friendly" strings (containing capital letters and space) with technical strings (containing underscores), and I would find it cleaner if all sub widgets were "within" their parents group (see also the comment above).

In other words, I would find it cleaner if the path for widgets would be:
`'state/{}/{}'.format(settings_key, widget_name)`
where settings_key was either an independent class variable (e.g. `settings_key = "array_editor"`) or derived from self.name (e.g. `settings_key = self.name.lower().replace(' ', '_')`).